### PR TITLE
Add dialog for adding serial and fetching the asset

### PR DESCRIPTION
**Pull request by zmwas**
**What does this PR do?**
Enables fetching asset data when serial number is entered.

**Description of Task to be completed?**
- Create serial entry dialog
- Fetch data from api using serial

**How should this be manually tested?**
Click on the check serial button and type a serial number when the dialog pops up . You will be redirected to the checkin activity

**Any background context you want to provide?**
Since the endpoint is not yet created you will have to use json web server and ngrok to make a fake web server then use the generated url as your base url.

**Screenshots**
screenshot_1522938091
screenshot_1522938098
screenshot_1522938110

### CODE REVIEW

File: app/src/main/res/layout/serial_dialog.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<layout
+    xmlns:android="http://schemas.android.com/apk/res/android">
+<android.support.design.widget.CoordinatorLayout
+    android:layout_width="360dp"
+    android:layout_height="1000dp"
```
Comment (suada-haji) [inline app/src/main/res/layout/serial_dialog.xml]: Place all these values in the `dimen` resource

File: app/src/main/res/layout/security_dashboard.xml [hunk 2]
```diff
@@ -1,68 +1,66 @@
...
+            android:layout_marginTop="@dimen/profile_button_margin_top"
+            android:background="@color/profileButtonBg"
+            android:fontFamily="@font/avenir_book"
+            android:paddingBottom="@dimen/profile_button_padding_top"
+            android:paddingLeft="@dimen/profile_button_padding_left"
+            android:paddingRight="@dimen/profile_button_padding_left"
+            android:paddingTop="@dimen/profile_button_padding_top"
+            android:textAllCaps="false"
+            android:text="@string/check_serial"
+            android:textColor="#ffffff"
```
Comment (suada-haji) [inline app/src/main/res/layout/security_dashboard.xml]: Use `android:textColor="@android:color/white"` instead

File: app/src/main/res/layout/serial_dialog.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    android:paddingLeft="24dp"
+    android:background="#ffffff"
```
Comment (suada-haji) [inline app/src/main/res/layout/serial_dialog.xml]: Use `android:background="@android:color/white"` instead

File: app/src/main/res/layout/serial_dialog.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        android:text="@string/enter_text"
+        android:background="#ffffff"
+        />
+
+    <EditText
+        android:id="@+id/serial_edit_text"
+        android:layout_width="278dp"
+        android:layout_height="50dp"
+        android:layout_marginTop="72dp"
+        android:backgroundTint="#35ccc8"
```
Comment (suada-haji) [inline app/src/main/res/layout/serial_dialog.xml]: Place this color in the resource folder

File: app/src/main/res/layout/serial_dialog.xml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    <Button
+        android:id="@+id/cancel"
+        android:layout_width="75dp"
+        android:layout_height="36dp"
+        android:background="@android:color/transparent"
+        android:layout_marginTop="144dp"
+        android:layout_marginBottom="16dp"
+        android:layout_marginLeft="115dp"
+        android:text="Cancel"
+        android:textColor="#35ccc8"
```
Comment (suada-haji) [inline app/src/main/res/layout/serial_dialog.xml]: This one too

File: app/src/main/res/layout/serial_dialog.xml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    <Button
+        android:id="@+id/submit"
+        android:layout_width="75dp"
+        android:layout_height="36dp"
+        android:background="@android:color/transparent"
+        android:layout_marginTop="144dp"
+        android:layout_marginLeft="207dp"
+        android:layout_marginBottom="16dp"
+        android:text="Submit"
+        android:textColor="#35ccc8"
```
Comment (suada-haji) [inline app/src/main/res/layout/serial_dialog.xml]: This too

File: app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <resources>
 
     <!-- Base application theme. -->
-    <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
+    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
```
Comment (suada-haji) [inline app/src/main/res/values/styles.xml]: Don't change this, it messes up with the theme for the Splash screen activity

Comment (zmwas) [reply app/src/main/res/values/styles.xml]: When I leave it as DarkActionBar I can not setSupportActionBar in the activity.

**Review by suada-haji — APPROVED** [review]
Looks good

**Pull request merged**